lib/reek/source/source_code.rb

```diff
@@ -37,9 +37,10 @@ def initialize(code, description, parser = Parser::Ruby22)
       # @return an instance of SourceCode
       def self.from(source)
         case source
-        when File   then new(source.read, source.path)
-        when IO     then new(source.readlines.join, 'STDIN')
-        when String then new(source, 'string')
+        when File     then new(source.read, source.path)
+        when IO       then new(source.readlines.join, 'STDIN')
+        when Pathname then new(source.read, source.to_s)
```

Comment:
The `String` option here is strictly for strings that contain the source code itself, so that is a different case.

How about making it the responsibility fo `.from`'s caller (i.e., `CLI::Input`) to convert the `Pathname` to a `File`?